app/(login)/CreateAccountScreen.tsx

```diff
@@ -0,0 +1,166 @@
+import styles from './styles'
+import { Text, View, } from '../../components/Themed';
+import { TextInput } from 'react-native-gesture-handler';
+import { ImageBackground, Pressable } from 'react-native';
+import { router } from 'expo-router';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+import { LogoAndName } from '../../components/LogoAndName';
+import { useState } from 'react';
+
+export default function CreateAccountScreen () {
+  const [phoneNumber, setPhoneNumber] = useState('')
+  const [password, setPassword] = useState('')
+  const [repeatPassword, setRepeatPassword] = useState('')
+  const [fullName, setFullName] = useState('')
+  const [email, setEmail] = useState('')
+  const [error, setError] = useState('')
+  const [showWrongPasswords, setShowWrongPasswords] = useState(false)
+  const [hidePassword, setHidePassword] = useState(true)
+  const [submitActive, setSubmitActive] = useState(false)
+  const [creatingUser, setCreatingUser] = useState(false)
+
+  function createAccount(ack : any) {
+    if (ack?.err){
+      setCreatingUser(false);
+      console.log("Some error on user creation")
+      if (ack.err == "User already created!"){
+          setError("User already exists")
+      }
+      else{//The other possible error is:  "User is already being created or authenticated!" which probably means the user has clicked create user multiple times
+        return
+      }
+    }
+    else{
+      let newUser = gun.user(ack.pub)
+      if (fullName != ''){
+        newUser.get('fullName').put(fullName)
+      }
+      if (email != '') {
+        newUser.get('email').put(email)
+      }           
+      setShowWrongPasswords(false)
+      user.auth(phoneNumber,password, login)
+    }
+  }
+
+  function login (ack : any) {            
+    if (!ack.err){    
+      if (user.is){            
+              console.log("Redirect to GroupScreen")
+              router.replace('/GroupScreen')            
+      }
+      else{
+          console.log("User somehow doesn't exist");
+      }
+    }
+    else{
+        console.log("Newly created user cannot be authenticated, something went wrong with the database"); 
+        //  Not sure how this should be handled
+    }
+  }
+
+  function validatePhone (phone : string) { 
```

Comment:
can be made shorter with `const validatePhone : boolean (phone : string) => phone.length == 8`